lib/internal/Magento/Framework/HTTP/Client/Curl.php

```diff
@@ -338,17 +338,18 @@ public function getStatus()
      * @param string $method
      * @param string $uri
      * @param array $params
+     * @param bool $json
      * @return void
      * @SuppressWarnings(PHPMD.CyclomaticComplexity)
      * @SuppressWarnings(PHPMD.NPathComplexity)
      */
-    protected function makeRequest($method, $uri, $params = [])
+    protected function makeRequest($method, $uri, $params = [], $json = false)
```

Comment:
Thank you for your contribution to Magento 2 project.

Adding this logic to protected method requires extension developers inherit this class. This is what we are trying to avoid in Magento 2.

Could you please provide more details about your use case? Maybe we will suggest a better way how we can achieve your goal